lib/manager/gradle-lite/tokenizer.ts

```diff
@@ -0,0 +1,231 @@
+import moo from 'moo';
+import { StringInterpolation, Token, TokenType } from './common';
+
+const escapedCharRegex = /\\['"bfnrt\\]/;
+const escapedChars = {
+  [TokenType.EscapedChar]: {
+    match: escapedCharRegex,
+    value: (x: string): string =>
+      ({
+        "\\'": "'",
+        '\\"': '"',
+        '\\b': '\b',
+        '\\f': '\f',
+        '\\n': '\n',
+        '\\r': '\r',
+        '\\t': '\t',
+        '\\\\': '\\',
+      }[x]),
+  },
+};
+
+export const rawLexer = {
+  // Top-level Groovy lexemes
+  main: {
+    [TokenType.Space]: { match: /[ \t\r]+/ },
```

Comment:
If you're using `\t` and `\r`, I'd use `\s` here too.

Also, is carriage return a valid space token?

Reply:
I didn't care about them too much, as we remove both spaces and newlines. To the another hand, I wonder if we can somehow detect and define explicit instruction separator marks, instead of `lookahead`s in parser level (which looks a bit hacky).